### PR TITLE
[KARAF-5454] - Collector socket - Add UDP protocol support

### PR DESCRIPTION
**Pull request by fpapon**
Improve the **collector-socket** adding compatibility with the UDP protocol.

Use a **DatagramServer** instead of a **ServerSocket** for UDP.

Changes :

- Add a new properties in the collector cfg file to define the protocol :
    `#Protocol tcp(default) or udp`
    `#protocol=tcp`

- Instanciate a DatagramServer in the case of udp protocol is define in the cfg
    class `org.apache.karaf.decanter.collector.socket.SocketCollector`

### CODE REVIEW

**Comment by jbonofre** [comment]
Thanks. I gonna take a look.

File: collector/socket/src/main/cfg/org.apache.karaf.decanter.collector.socket.cfg [hunk 1]
```diff
@@ -4,4 +4,7 @@
 #port=34343
 
 # Number of worker threads to deal with
-#workers=10
\ No newline at end of file
+#workers=10
+
+# Protocol tcp(default) or udp
+#protocol=tcp
```
Comment (jbonofre) [inline collector/socket/src/main/cfg/org.apache.karaf.decanter.collector.socket.cfg]: Please avoid to do useless changes (it's just noise in the commit ;)).

**Comment by jbonofre** [comment]
Any update about this PR ?

**Comment by fpapon** [comment]
Sorry, I check this

**Comment by jbonofre** [comment]
retest this please

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/karaf-decanter-pr/1/

**Comment by jbonofre** [comment]
retest this please

**Comment by asfgit** [comment]
FAILURE
 
--none--

**Comment by jbonofre** [comment]
Failure doesn't seem related as it's on the JMS collector. Launching a new build to verify.

**Comment by jbonofre** [comment]
retest this please

**Comment by asfgit** [comment]
SUCCESS 
 
--none--

**Comment by jbonofre** [comment]
Build is now OK.

**Pull request merged**